encoder/encoder.hpp

```diff
@@ -33,6 +33,8 @@ class Encoder
 	// describing a DMABUF, and by a mmapped userland pointer.
 	virtual void EncodeBuffer(int fd, size_t size, void *mem, StreamInfo const &info, int64_t timestamp_us) = 0;
 
+	virtual void Signal() {}
```

Comment:
Perhaps a comment could be added above to explain what this method is for.

Reply:
I don't want to be too nit-picky, but all the other comments appear above their respective function declarations.

I wonder if there's some way the wording could be a little more general/abstract, so that other Encoder sub-class writers will know what to do (or why they might or might not need to implement it)?

Reply:
Agree, probably just move the comments before the declaration.